R/rdeephaven/tests/testthat/test_client_wrapper.R

```diff
@@ -0,0 +1,204 @@
+    expect_no_error(client$import_table(as_record_batch_reader(arrow_table(data$df3))))
+    expect_no_error(client$import_table(as_record_batch_reader(arrow_table(data$df4))))
+})
+
+# The following tests additionally assume the correctness of client$import_table(...) AND table_handle$bind_to_variable(),
+# as we have to create data, push it to the server, and name it in order to test client$open_table().
+# Additionally, we assume the correctness of table_handle$to_data_frame() to make concrete comparisons.
+
+test_that("open_table opens the correct table from the server", {
+    data <- setup()
+
+    client_options <- ClientOptions$new()
+    client <- Client$new(target="localhost:10000", client_options=client_options)
+
+    th1 <- client$import_table(data$df1)
+    th1$bind_to_variable("table1")
+    expect_equal(client$open_table("table1")$to_data_frame(), th1$to_data_frame())
+
+    th2 <- client$import_table(data$df2)
+    th2$bind_to_variable("table2")
+    expect_equal(client$open_table("table2")$to_data_frame(), th2$to_data_frame())
+
+    th3 <- client$import_table(data$df3)
+    th3$bind_to_variable("table3")
+    expect_equal(client$open_table("table3")$to_data_frame(), th3$to_data_frame())
+
+    th4 <- client$import_table(data$df4)
+    th4$bind_to_variable("table4")
+    expect_equal(client$open_table("table4")$to_data_frame(), th4$to_data_frame())
+})
+
+test_that("run_script correctly runs a python script", {
+
+    client_options <- ClientOptions$new()
+    client <- Client$new(target="localhost:10000", client_options=client_options)
+
+    expect_no_error(client$run_script(
+'
+from deephaven import new_table
+from deephaven.column import string_col, int_col
+
+static_table_from_python_script = new_table([
+string_col("Name_String_Col", ["Data String 1", "Data String 2", "Data String 3"]),
+int_col("Name_Int_Col", [44, 55, 66])
+])
+'
+    ))
+
+    expect_no_error(client$open_table("static_table_from_python_script"))
+})
+
+##### TESTING BAD INPUTS #####
+
+test_that("client connection fails nicely with bad target but good client_options", {
+
+    # assumes correctness of client options
+    client_options <- ClientOptions$new()
+
+    # TODO: Bad address needs better error handling from the R side
+    expect_error(client <- Client$new(target="bad address", client_options=client_options))
+    expect_error(client <- Client$new(target=12345, client_options=client_options),
+        "'target' must be passed as a single string. Got object of class numeric instead.")
+    expect_error(client <- Client$new(target=c("hello", "my", "name", "is"), client_options=client_options),
+        "'target' must be passed as a single string. Got character vector of length 4 instead.")
+})
+
+test_that("client connection fails nicely with good target but bad client_options", {
+
+    # TODO: these all assume that the server is actually running on localhost:10000, probably bad for CI
```

Comment:
todo

Reply:
Feels like a cross-that-bridge when we get to CI. No idea how Colin wants to set that up yet.